### PR TITLE
Added Havel-Hakimi.cpp

### PR DESCRIPTION
**Pull request by urvashiramdasani**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

I have added the Havel-Hakimi algorithm that checks whether a degree sequence is graphical or not. I have implemented in C++ as a part of GirlScript Summer of Code (GSSoC) 2021.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue #3965

### CODE REVIEW

File: C-Plus-Plus/graphs/Havel-Hakimi.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+4
+4
+3
+2
+1
+-1
+
+The degree sequence is graphical
+
+*/
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/Havel-Hakimi.cpp]: Use multi-line comments and add a new line at the EOF

File: C-Plus-Plus/graphs/Havel-Hakimi.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
+// Use Havel-Hakimi theorem and check whether the given degree sequence is graphical or not.
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/Havel-Hakimi.cpp]: Add more information about the algorithm

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by urvashiramdasani** [comment]
Hey @HarshCasper, please check the code now. Thanks!

File: C-Plus-Plus/graphs/Havel-Hakimi.cpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+1. All the elements remaining are equal to 0 (Simple graph exists).
+2. Negative number encounter after subtraction (No simple graph exists).
+3. Not enough elements remaining for the subtraction step (No simple graph exists).
+
+*/
+
+// Include all libraries
+#include <iostream>
+#include <vector>
+#include <algorithm>
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/graphs/Havel-Hakimi.cpp:25]: Replace header files with `#include <bits/stdc++.h>`

File: C-Plus-Plus/graphs/Havel-Hakimi.cpp [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+		v.erase(it);
+
+		// Resorting
+		sort(v.begin(), v.end(), greater<int>());
+
+		// Check if vector contains all zeros
+		if(all_of(v.begin(), v.end(), [](int i) { return i == 0; })) return true;
+
+		// Return false if any element becomes -1
+		if(all_of(v.begin(), v.end(), [](int i) { return i == -1; })) return false;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/graphs/Havel-Hakimi.cpp:52]: Use proper code Format

File: C-Plus-Plus/graphs/Havel-Hakimi.cpp [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+4
+4
+3
+2
+1
+-1
+
+The degree sequence is graphical
+
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/graphs/Havel-Hakimi.cpp:97]: Remove this redundant part and ,
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
Closing this due to inactivity.

**Pull request closed without merging**